### PR TITLE
Adding first how-to guide as a basic example

### PR DESCRIPTION
**Pull request by jimbobbennett**
Created a first how-to guide for saving a snippet to show the basic structure.

Also added a contributing file with more details.

I've added the `.vscode` folder back to include settings for CSpell - the [code spell checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker). This sets things like enabling it for MDX files, and allows us to ingore certain words.

### CODE REVIEW

**Review by shivay-at-pieces — COMMENTED** [review]
I think all of the assets should go into the assets directory and only the .mdx file should be there in the docs section.

File: docs/how-to-guides/save-a-snippet/save-a-snippet.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+<div class="container">
+    <div class="row">
+        <div class="col col--8">
+            ![The saved materials tab](saved-materials-tab.webp)
+        </div>
+    </div>
+</div>
+
+### Add the code snippet
```
Comment (shivay-at-pieces) [inline docs/how-to-guides/save-a-snippet/save-a-snippet.mdx:45]: @jimbobbennett would love to know your thoughts: 

We can add a small code snippet here in the how-to-guide as well for quick access and ask the user to copy that snippet and paste it, if they don't have a snippet handy.

Comment (jimbobbennett) [reply docs/how-to-guides/save-a-snippet/save-a-snippet.mdx:45]: Good idea - let me add that.

Comment (jimbobbennett) [reply docs/how-to-guides/save-a-snippet/save-a-snippet.mdx:54]: Added - both code inline and a downloadable file

**Review by shivay-at-pieces — COMMENTED** [review]
Looks good!

**Comment by mason-at-pieces** [comment]
@jimbobbennett @shivay-at-pieces So, about the assets being in the docs directory.

I agree that global assets are great to have in the assets folder. I'm fine with having one-off assets in their own respective folders. The `assets` folder has become quite messy just because it was started as a centralized location for storage.

For media that are used in multiple places, let's put them in the global `static` folder. For media that is specific to a page, put it inside a `media` folder. Inside that `media` folder, please also make folders for `images`, `videos`, etc so that is remains organized and not just a folder full of different types of files.

**Review by mason-at-pieces — CHANGES_REQUESTED** [review]
Please move all `webp` files into a `media/images` folder so we keep the directory clean.

Also please move the `helloworld.py` file into a folder called `resources`

**Comment by shivay-at-pieces** [comment]
@mason-at-pieces you can have a look at the updated structure for the resources and media

**Pull request merged**